### PR TITLE
Transfer learning

### PR DESCRIPTION
**Pull request by SinaSarparast**
**Main Contributions**
- [x] prototyping using ResNet18
- [x] add few shot learning dataloader and sampler for Office31 dataset
- [x] prototyping using FSL framework (ProtoNet) and ResNet18

**Bug Fixes**
- Corrected wrong dimension in the Softmax function of the ResNet module.
- Corrected the path specifying Office31 dataset's location.

### CODE REVIEW

File: image_classification_simulation/data/office31_loader.py [hunk 1]
```diff
@@ -33,17 +33,30 @@ def __init__(
         super().__init__(data_dir, hyper_params)
         self.num_unique_labels = 31
         hyper_params["num_classes"] = self.num_unique_labels
-        self.n_way = hyper_params["n_way"]
-        self.n_shot = hyper_params["n_shot"]
-        self.n_query = hyper_params["n_query"]
-        self.num_training_episodes = hyper_params["num_training_episodes"]
-        self.num_eval_tasks = hyper_params["num_eval_tasks"]
-
+        if "n_way" in hyper_params:
```
Comment (SinaSarparast) [inline image_classification_simulation/data/office31_loader.py:36]: I think these checks should be done elsewhere.

**Review by alzaia — APPROVED** [review]
I agree about having too many checks being done on the dataloader classes, but let's approve them like this right now and add an issue to refactor these in a better way so we do not forget about it.
